packages/vscode-extension/src/devices/AndroidEmulatorDevice.ts

```diff
@@ -301,15 +301,20 @@ export class AndroidEmulatorDevice extends DeviceBase {
     if (build.platform !== DevicePlatform.Android) {
       throw new Error("Invalid platform");
     }
-    // adb install sometimes fails because we call it too early after the device is initialized.
-    // we haven't found a better way to test if device is ready and already wait for boot_completed
-    // flag in waitForEmulatorOnline. But even after that even is delivered, adb install also sometimes
-    // fails claiming it is too early. The workaround therefore is to retry install command.
-    if (forceReinstall) {
+
+    // allowNonZeroExit is set to true to not print errors when INSTALL_FAILED_UPDATE_INCOMPATIBLE occurs.
+    const installApk = (allowDowngrade: boolean) =>
+      exec(
+        ADB_PATH,
+        ["-s", this.serial!, "install", ...(allowDowngrade ? ["-d"] : []), "-r", build.apkPath],
```

Comment:
any reason why we can't just always "allow downgrade" ??

Reply:
`adb` only allows downgrading debuggable packages. If we assume that every app we install is debuggable then we can always do it.

Reply:
that's a good point. I think it's ok to assume process is debuggable unless it's expo go